src/sinks/datadog/logs/tests.rs

```diff
@@ -230,7 +303,30 @@ async fn multiple_api_keys() {
 }
 
 #[tokio::test]
-async fn enterprise_headers() {
+/// Assert that events are sent and the DD-EVP-ORIGIN header is set when
+/// 'enterprise' is flagged on, v2 API
+///
+/// Vector allows for flagging a global 'enterprise' context that indicates
+/// whether we're running in Datadog enterprise mode or not. When this flag is
+/// active we should set the origin header discussed above correctly, as well as
+/// still sending events through the sink.
+async fn enterprise_headers_v2() {
+    enterprise_headers_inner(ApiStatus::OKv2).await
+}
+
+#[tokio::test]
+/// Assert that events are sent and the DD-EVP-ORIGIN header is set when
+/// 'enterprise' is flagged on, v1 API
+///
+/// Vector allows for flagging a global 'enterprise' context that indicates
+/// whether we're running in Datadog enterprise mode or not. When this flag is
+/// active we should set the origin header discussed above correctly, as well as
+/// still sending events through the sink.
+async fn enterprise_headers_v1() {
+    enterprise_headers_inner(ApiStatus::OKv1).await
+}
```

Comment:
There's no links to documentation about the header in the implementation so that's a good call out. I dunno, though it does function. https://github.com/vectordotdev/vector/issues/9298 doesn't discuss API versions.

Reply:
I _think_ we passed a header before with vector + version that we've used internally, I don't know if the new headers might cause backward compatibility problems...

cc @jdrouet @vladimir-dd

Reply:
these headers will be simply ignored by v1 endpoints, so it is backward-compatile.